docs/libraries/python/examples.md

```diff
@@ -1,5 +1,538 @@
+ ('atoi1qq6lkr9hucfylqjaqphu0stvk8pcmsx98r7ukuq40asszwmqytlnc058thk', True),
+ ('atoi1qzpn7se3ryhscmqg404pycxzvfpt8v4xn8aul0tqdh00xsncgnxu7na7zjj', False),
+ ('atoi1qz4qqakty9qytw8fk9shelt9lwlvv83s5ggt3wjag9fkgcc74z78w4l86y5', True),
+ ('atoi1qp20uddchglqry0l5qnjg5aln8d5rk2v5l45hwrxv9z0daxs7u6xcsh4077', False),
+ ('atoi1qrlqm2u5txxxnjx22fxq0jfjzk6l4nwnue6ht5pepk65m2f4xmxqynmxu2m', True),
+ ('atoi1qqydc70mpjdvl8l2wyseaseqwzhmedzzxrn4l9g2c8wdcsmhldz0ulwjxpz', False),
+ ('atoi1qrkjennxyl2xcqem6x69ya65sasma33z0ux872k846lqft0s3qf7k6lqpft', True),
+ ('atoi1qr4yuekp30ff7mnnnjwy9tdhynxmlmkpuxf70qurtwudp2zpf3jeyw4uh37', False),
+ ('atoi1qp6m5sz5ayjtccfxapdk5lp4qkheyfg0emzntmulyxzftps730vcul8dmqr', True),
+ ('atoi1qzrwhkzhu67fqltfffwljejawdcghedukpgu9x6tzevwlnq89gmfjtayhgz', False),
+ ('atoi1qpehxcp24z947dgupjqc9ktkn5ylmdxqqnx83m7xlajnf8005756u4n7z77', True)]
+```
+* Each tuple contains `address` and `bool` value indicating the given address is a `change` address or not.<br />
+`True` means the given address is a change address (internal). So basically we've got two independent sets of addresses (10 items per each)
+* This behavior is controlled via `get_all` argument. `get_all=False` (default) means to generate only public addresses
+
+IOTA address is represented by a checksumed base 32 string (Bech32) and you can see a detailed explanation on [Chrysalis docs](https://chrysalis.docs.iota.org/guides/index.html#iota-15-address-anatomy).
+Just a recap:
+* If an address starts with `atoi` then it means it is related to `testnet`. `iota` stands for mainnet
+* Number `1` at 5<sup>th</sup> position is just a separator
+* The last 6 characters are reserved for a checksum
+
+To quickly validate any IOTA address, there is a convenience function `Client.is_address_valid()` that returns `bool` value. Needless to say, performing a sanity check of an address before its use is an advisable practice.
+
+## Checking a balance
+_In Chrysalis testnet, there is a faucet service that provides test tokens to any testnet address: https://faucet.testnet.chrysalis2.com/_
+
+There are three common api calls that can be leveraged:
+* `Client.get_address_balance(address: str)`: it expects a single address in Bech32 format and returns `dict` with a balance for the address
+* `Client.get_address_balances(list[str])`: a convenience function that expects `list` of addresses in Bech32 format and returns list of `dict` with balances for all given addresses
+* `Client.get_balance(seed, account_index (optional), initial_address_index(optional), gap_limit(optional))`: a convenience function that combines `Client.get_addresses()` and `Client.get_address_balances()` api calls. It returns a combined balance for the provided seed and its wallet account index
+
+_Please note: `Client.get_address_balance()` and `Client.get_address_balances()` return address(es) in hex-encoded Ed25519 address format, which is the format returned by underlying node software:_
```

Comment:
Maybe we should convert it to a bech32 address, what do you think @bingyanglin @Hribek25 ?

Reply:
> Maybe we should convert it to a bech32 address, what do you think @bingyanglin @Hribek25?
> The same for the nodejs binding

I think it is awesome idea. Plus, maybe we can also add functions for conversion from hex to bech32 and vice versa. If I am not mistaken, those functions are not avail in bindings.

Reply:
Good point. I will do this.

Reply:
@bingyanglin @melatron already started #467, but maybe you guys can split the work